src/init/pkg_installer.sh

```diff
@@ -129,86 +123,46 @@ if [ -z "${WAZUH_REVISION}" ] ; then
-declare -a FOLDERS_TO_BACKUP
-
-echo "$(date +"%Y/%m/%d %H:%M:%S") - Generating Backup." >> ./logs/upgrade.log
-
-# Generate wazuh home directory tree to backup
-[ -d "./active-response" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/active-response)
-[ -d "./bin" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/bin)
-[ -d "./etc" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/etc)
-[ -d "./lib" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/lib)
-[ -d "./queue" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/queue)
-[ -d "./ruleset" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/ruleset)
-[ -d "./wodles" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/wodles)
-[ -d "./agentless" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/agentless)
-[ -d "./logs/ossec" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/logs/ossec)
-[ -d "./var/selinux" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/var/selinux)
-
-for dir in "${FOLDERS_TO_BACKUP[@]}"; do
-    mkdir -p "${TMP_DIR_BACKUP}${dir}"
-    rsync -a --exclude "diff" ${dir}/* "${TMP_DIR_BACKUP}${dir}"
-done
-
-if [ -f $OSSEC_INIT_FILE ]; then
-    mkdir -p "${TMP_DIR_BACKUP}/etc"
-    cp -p $OSSEC_INIT_FILE "${TMP_DIR_BACKUP}/etc"
-fi
-
 # Check if systemd is used
 # RHEL 8 >= services must must be installed in /usr/lib/systemd/system/
 if [ -f /usr/lib/systemd/system/${SERVICE}.service ] && [ ! -h /usr/lib/systemd/system ]; then
     SYSTEMD_SERVICE_UNIT_PATH=/usr/lib/systemd/system/${SERVICE}.service
-    mkdir -p "${TMP_DIR_BACKUP}/usr/lib/systemd/system/"
-    cp -a "${SYSTEMD_SERVICE_UNIT_PATH}" "${TMP_DIR_BACKUP}${SYSTEMD_SERVICE_UNIT_PATH}"
 fi
 # Others
 if [ -f /etc/systemd/system/${SERVICE}.service ] && [ ! -h /etc/systemd/system ]; then
     SYSTEMD_SERVICE_UNIT_PATH=/etc/systemd/system/${SERVICE}.service
-    mkdir -p "${TMP_DIR_BACKUP}/etc/systemd/system/"
-    cp -a "${SYSTEMD_SERVICE_UNIT_PATH}" "${TMP_DIR_BACKUP}${SYSTEMD_SERVICE_UNIT_PATH}"
 fi
 
 # Init backup
 # REHL <= 6 / Amazon linux
 if [ -f "/etc/rc.d/init.d/${SERVICE}" ] && [ ! -h /etc/rc.d/init.d ]; then
     INIT_PATH="/etc/rc.d/init.d/${SERVICE}"
-    mkdir -p "${TMP_DIR_BACKUP}/etc/rc.d/init.d/"
-    cp -a "${INIT_PATH}" "${TMP_DIR_BACKUP}${INIT_PATH}"
 fi
 
 if [ -f "/etc/init.d/${SERVICE}" ] && [ ! -h /etc/init.d ]; then
     INIT_PATH="/etc/init.d/${SERVICE}"
-    mkdir -p "${TMP_DIR_BACKUP}/etc/init.d/"
-    cp -a "${INIT_PATH}" "${TMP_DIR_BACKUP}${INIT_PATH}"
 fi
 
-# Saves modes and owners of the directories
-BACKUP_LIST_FILES=$(find "${TMP_DIR_BACKUP}/" -type d)
+# Generate Backup
+BDATE=$(date +"%m-%d-%Y_%H-%M-%S")
+mkdir -p ./backup
 
-while read -r line; do
-    org=$(echo "${line}" | awk "sub(\"${TMP_DIR_BACKUP}\",\"\")")
-    chown --reference=$org $line >> ./logs/upgrade.log 2>&1
-    chmod --reference=$org $line >> ./logs/upgrade.log 2>&1
-done <<< "$BACKUP_LIST_FILES"
+echo "$(date +"%Y/%m/%d %H:%M:%S") - Generating Backup." >> ./logs/upgrade.log
 
+FOLDERS_TO_BACKUP=($WAZUH_HOME/{active-response,bin,etc,lib,queue,ruleset,wodles,agentless,logs/{ossec,wazuh},var/selinux} \
```

Comment:
Here it's including `logs/wazuh` and `logs/ossec` but not `logs`.

Reply:
This is OK as including `logs` would delete the entire folder on rollback, including `logs/upgrade.log`.